crates/matrix-sdk/src/widget/machine/from_widget.rs

```diff
@@ -245,10 +254,23 @@ impl From<update_delayed_event::unstable::Response> for UpdateDelayedEventRespon
     }
 }
 
-/// The response to the widget that it received the to-device event.
-/// Only used as the response for the successful send case.
-/// FromWidgetErrorResponse will be used otherwise.
-/// This is intentionally an empty tuple struct (not a unit struct), so that it
-/// serializes to `{}` instead of `Null` when returned to the widget as json.
-#[derive(Serialize, Debug)]
-pub(crate) struct SendToDeviceEventResponse {}
+/// Send to device request response.
+#[derive(Serialize, Debug, Default)]
+pub(crate) struct SendToDeviceEventResponse {
+    /// It is possible that sending to some device failed (failed to encrypt or
+    /// failed to send)
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub failures: Option<BTreeMap<String, Vec<String>>>,
```

Comment:
For now there is no reason/code for the failures. Will be handled in a follow-up PR